src/nvim/assert.h

```diff
@@ -121,4 +121,33 @@
     ((enum { ASSERT_CONCAT(assert_line_, __LINE__) = 1/(!!(e)) }) 0)
 #endif
 
+/// @def STRICT_ADD
+///
+/// Requires GCC 5+ and Clang 3.8+
+///   https://clang.llvm.org/docs/LanguageExtensions.html
+///   https://gcc.gnu.org/onlinedocs/gcc/Integer-Overflow-Builtins.html
+///
+/// Alternative for compilers without __builtin_xx_overflow ?
+///   https://stackoverflow.com/a/44830670/152142
+///
+/// @param MAX Maximum value of the narrowest type of operand.
+///            Not used if compiler supports __builtin_add_overflow.
+#if (defined(__clang__) && __has_builtin(__builtin_add_overflow)) \
+  || (__GNUC__ >= 5)
```

Comment:
gcc does not support `__has_builtin`. Is there another way to feature-detect `__builtin_add_overflow` on gcc?

Reply:
Use CMake's `check_c_source_compiles()` instead?  That would work for both compilers.